.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,65 @@
+name: Build and publish
+on: [push]
+version: 2
+jobs:
+  build:
+    runs-on: [self-hosted, Linux, AWS]
+    timeout-minutes: 10
+    steps:
+      - name: Persist Workspace
+        uses: Brightspace/third-party-actions@actions/cache
+        with:
+          path: |
+            ./*
+          key: worksapce-v1-${{ github.sha }}
+
+      - name: Checkout
+        uses: Brightspace/third-party-actions@actions/checkout
+
+      - run:
+          name: Authenticate with Artifactory
+          command: curl -ks -u$DOCKER_USER:$DOCKER_PASS "https://d2lartifacts.jfrog.io/artifactory/api/npm/npm-local/auth/d2l" -o ~/project/.npmrc
```

Comment:
you probably do not need to do this anymore since you are not pushing things to jfrog